trader/src/parse_system.rs

```diff
@@ -37,16 +39,49 @@ pub fn parse_json() -> Result<Vec<System>, std::io::Error> {
             let mut contents = String::new();
             file.read_to_string(&mut contents)
                 .expect("Failed to read file");
-
             // Deserialize the JSON into a Vec<System>
             let systems: Vec<System> =
                 serde_json::from_str(&contents).expect("Failed to parse JSON");
             Ok(systems)
         }
         Err(error) => {
             println!("Failed to open file: {}", error);
-
+            let download_result = download_file(
+                "https://api.spacetraders.io/v2/systems.json",
+                "./config/systems.json",
+            );
+            match download_result {
+                Ok(_) => {
+                    println!("File downloaded successfully");
+                    let mut file = File::open(file_path).unwrap();
+                    let mut contents = String::new();
+                    file.read_to_string(&mut contents)
+                        .expect("Failed to read file");
+                    let systems: Vec<System> =
+                        serde_json::from_str(&contents).expect("Failed to parse JSON");
```

Comment:
These two expects should probably be replaced with a `.wrap_err("")?` operator

change the type the function is returning to `color_eyre::Result<Vec<System>>` and it should allow you to return any error types (color_eyre is like anyhow if you know that crate)